### PR TITLE
Framework: Pull in i18n-calypso (alt lyrics)

### PR DESCRIPTION
**Pull request by ockham**
Alternative to #27123

Using `lerna import` this time, to preserve commit history :heart_eyes: 

Note that we'll have to do this over, since the i18n-calypso repo has since diverged. We can stiill use this as a playground to sort out things.

TODO
- [ ] Properly set up testing/CI
- [ ] Provide CommonJS build so we can revert 02e873ad826f03210e88964216c5566171450980. Alternatively/additionally, use https://www.npmjs.com/package/esm?
  - We might want to provide CommonJS anyway, so that 3rd parties can easily consume the npm.

### Testing instructions
- `npm run distclean`
- `npx lerna bootstrap`
- `npm start`

### CODE REVIEW

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+	// Should catch template literals
+	content = i18n.translate( `My hat has six corners.` );
+	content = i18n.translate( `My hat
+has seventeen
+corners.` );
+
+	// providing context
+	content = i18n.translate( {
+		original: 'post',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This string already exists with the following **contexts**:
- `null` (no context)
- `add new from admin bar`
- `noun`
- `post a comment`
- `post type singular name`
- `Verb, to post`

 Would it make sense to reuse one of the above?

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		context: 'verb'
+	} );
+
+	// passing string as initial argument
+	content = i18n.translate( 'post2', { context: 'verb2' } );
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	var varCount = 1;
+	content = i18n.translate( 'single test2', 'plural test2', { count: varCount } );
+
+	// test the ability to break strings to multiple lines
+	content = <div>{ i18n.translate(
+		"This is a multi-line translation with \"mixed quotes\" " +
+		'and mixed \'single quotes\''
+	) }</div>;
+
+	content = this.translate( 'The string key text',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+		{ 'context': 'context with a literal string key' } );
+
+	i18n.translate( 'My hat has three corners.', {
+		'comment': 'Second ocurrence'
+	} );
+
+	i18n.translate( 'My hat has one corner.', 'My hat has many corners.', {
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/cli/examples/i18n-test-examples.jsx]: :information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

File: packages/i18n-calypso/test/index.js [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+					'': data.locale[''],
+					'test1': [
+						null,
+						'translation1-1'
+					],
+					'test2': [
+						null,
+						'translation2'
+					],
+					'new translation': [
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: Hi! I've found a possible matching string that has already been translated **2** times:
`translate( 'New Translation Set' )` *ES Score: 10*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: packages/i18n-calypso/test/index.js [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+			} );
+			it( 'should fall back to original if translation isn\'t even null in locale file', function() {
+				assert.equal( 'nonexisting-string', translate( 'nonexisting-string' ) );
+			} );
+		} );
+
+		describe( 'translate with context', function() {
+			it( 'should find a string with context', function() {
+				assert.equal( 'translation3', translate( {
+					original: 'test3',
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: :information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

File: packages/i18n-calypso/test/index.js [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+				'': {
+					localeSlug: 'xx-pig-latin',
+					'key-hash': 'sha1'
+				},
+				'0f7d0d088b6ea936fb25b477722d734706fe8b40': [
+					null,
+					'implesa'
+				]
+			});
+			assert.equal( i18n.translate( 'simple' ), 'implesa' );
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: Hi! I've found a possible matching string that has already been translated **35** times:
`translate( 'Simple', { context: 'Theme Showcase term name'} )` *ES Score: 14*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: packages/i18n-calypso/test/index.js [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+				'': {
+					localeSlug: 'xx-pig-latin',
+					'key-hash': 'sha1-1-2'
+				},
+				'0': [ null,'implesa' ],
+				'78': [ null, 'edra' ],  // red has a sha1 of 78988010b890ce6f4d2136481f392787ec6d6106
+				'7d': [ null, 'reyga' ] // grey has a sha1 of 7d1f8f911da92c0ea535cad461fd773281a79638
+			});
+			assert.equal( i18n.translate( 'simple' ), 'implesa' );
+			assert.equal( i18n.translate( 'red' ), 'edra' );
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: Hi! I've found a possible matching string that has already been translated **104** times:
`translate( 'Red' )` *ES Score: 12*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/i18n-repo-take-two)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: packages/i18n-calypso/test/index.js [hunk 9]
```diff
@@ -0,0 +1,376 @@
...
+			assert.equal( i18n.translate( 'grey' ), 'reyga' );
```
Comment (a8ci18n) [inline packages/i18n-calypso/test/index.js]: Hi! I've found a possible matching string that has already been translated **52** times:
`translate( 'Grey' )` *ES Score: 16*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by sirreal** [comment]
Mentioned elsewhere (https://github.com/Automattic/wp-calypso/pull/27356#issuecomment-425393820), but I'd like to experiment with [`projects`](https://jestjs.io/docs/en/configuration.html#projects-array-string-projectconfig) to get package tests running.

**Comment by ockham** [comment]
Still can't get this to bundle properly. Tried adding to `resolve.modules`, tried aliasing, tried changing `main` to `module` in `packages/i18n-calypso/package.json` (props @blowery) -- always getting

```
SyntaxError: Unexpected token export
```

**Comment by blowery** [comment]
Ahhhh.. it's used server side. We mark everything in `node_modules` as an external in the server side build so it doesn't get pulled into the bundle, it ends up getting pulled in using a normal require call when the server boots up.

You'll need to provide _two_ variants of i18n-calypso. One with a package.json::main that points to an es5 + common JS export style for the server build and one with package.json::module that points to es5 + es6 export style for the client-side webpack build.

**Pull request closed without merging**